### DISCUSSION TITLE
V3 Plan

### DISCUSSION THREAD

**Discussion by RetricSu** [nervosnetwork/docs.nervos.org#347] [Ideas]
See https://github.com/nervosnetwork/docs.nervos.org/discussions/347#discussioncomment-9643380 and https://github.com/nervosnetwork/docs.nervos.org/discussions/347#discussioncomment-9643611
----
We plan to ship a V3 version of the doc site before the end of June. 

What features/content should be included in the V3 version?

Some ideas:

-  [ ] Comparison information: Bitcoin and CKB differences
-  [ ] ~(optional) Overview page(guideline of the doc site, like https://ethereum.org/en/developers/docs/) bad idea?~
-  [ ] All in one page: Link to all resources ( https://academy.ckb.dev/ https://cookbook.ckbdapps.com/ etc)
-  [ ] Better UX/UI
   - [ ] Tooltips for terms on each page
-  [ ] Common Scripts 
    - [ ] How this Script works
    - [ ] Script deployed info
    - [ ] Offchain SDK examples
    - [ ] On-chain integration examples
  - [ ] Wallets
    - [ ] CCC integration tutorials
    - [ ] `offckb` template integrated with wallets
    - [ ] New supported wallets update
  - [ ] Enhance Visual Accessibility for tech explanation https://github.com/nervosnetwork/docs.nervos.org/issues/240
     - [ ] Enhance `How CKB works` with intuitive concept images (Cell box vs UTXO box)
     - [ ] Concept in Tech explanation with flowchart
  - [ ] SDK Section
  - [ ] How-to Section
  - [ ] Molecule Section
  - [ ] RFC listed
     - [ ] Chat with LLM for RFC?

**Comment by RetricSu** [comment]
Principle:
- https://docs.divio.com/documentation-system/

Some docs site examples worth learning:

- https://learnmeabitcoin.com/
- https://ethereum.org/en/developers/docs/

**Comment by yfeng2824** [comment]
> (optional) Overview page(guideline of the doc site, like https://ethereum.org/en/developers/docs/) bad idea?

I don't think we need an overview page since our current landing page already serves that purpose. Instead, we can analyze user behavior and adjust our starting points to better meet their needs.

---
Besides the ideas shared above, I thought we could add tooltips for terms on each page. This would give users instant explanations without needing to navigate away, making the content more accessible and saving time. We can use the first sentence from our `glossary.md` for the tooltip content and include a "learn more" link for deeper exploration on the glossary or the relevant page.
It might look something like this: 
<img width="515" alt="Screenshot 2024-05-21 at 12 46 59 PM" src="https://github.com/nervosnetwork/docs.nervos.org/assets/140578792/ad062d64-2eef-4b17-b268-2e496bda4064">

**Reply by RetricSu** [reply]
yeah i like this idea too, ckb academy have something like this https://academy.ckb.dev/courses/basic-theory
![image](https://github.com/nervosnetwork/docs.nervos.org/assets/23436060/fb3a396f-adbc-46eb-84ff-cc22c4bd4610)

**Comment by linnnsss** [comment]
A few questions about the arrangement of the tentative categories:

1. For serialization, maybe better to place it under 'how-to' or 'tech explanation'? But if it is a core and frequently encountered component, keeping it as a top-level category is fine.

2. Similarly, for 'common scripts', placing it under 'Scripts (Smart Contracts)' might be preferable. If this makes the structure too deep, keeping it directly under Scripts would be fine.

3. For the 'How-tos', we've already gathered a list of topics, or do we need to collect them from the developers?

**Reply by RetricSu** [reply]
1. yea it is a core component
2. I feel like it will be too deep
3. we have very few topics, need to collect

**Comment by RetricSu** [comment]
### Doc V3 Plan
- [ ] New content
  -  [ ] Comparison: Bitcoin and CKB differences @RetricSu @yfeng2824 
      - Stack vs CKB-VM
      - UTXO vs Cell
      - Transaction Structure
      - POW
  
  - [x] Dapp Section @RetricSu 
      - [x] full-stack simple lock example
      - [ ] (optional) full-stack dex example
      
      
  - [x] Script Section
      - [x] migrate xuejie's blog posts @RetricSu 
      - [x] CKB syscall for Script (based on [RFC](https://github.com/XuJiandong/rfcs/blob/master/rfcs/0046-syscalls-summary/0046-syscalls-summary.md) ) @linnnsss 
      - [x] Script Cycles Limits (based on RFC)
      - [ ] (optioanl) Compiling Script (based on Script toolchain maintaince report)

  -  [x] Common Scripts(Omnilock/Xudt) @RetricSu @linnnsss 
      - [x] How this Script works @RetricSu 
      - [x] Script deployed info @RetricSu 
      - [x] Offchain SDK examples @linnnsss 
      - [ ] (optional) On-chain integration examples
      
  - [x] SDK Section(Rust/Java/Javascript/Typescript/Other Devtools) @yfeng2824 
      - [x] Introduce(suitable for what kind of situation)
      - [x] Requirements
      - [x] Examples

  - [x] Integrate Wallets @RetricSu 
    - [x] CCC integration tutorials
    - [x] `offckb` template integrated with wallets
    - [x] New supported wallets update if any
 
  - [x] Molecule Section
      - [x] Why do I need this?
      - [x] How Molecule Works @linnnsss based on https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0008-serialization/0008-serialization.md
      - [ ] (optional) Offchain Examples
      - [ ] (optioanl) OnChain Examples
  
  - [x] Ecosystem Section
      - [x] github orgs @yfeng2824 
  
  - [x] Resource Links @linnnsss 
     - https://academy.ckb.dev/ 
     - https://cookbook.ckbdapps.com/
     - RGB++
     - Spore
     - Other useful links
  
  - [x] How-to Section
      - [x] https://github.com/nervosnetwork/ckb-system-scripts/wiki/How-to-sign-transaction @linnnsss 
      - [x] (optional) how to calculate all kinds of hashes
           - specs
           - ~~SDKs & tools~~
           
  - [x] Tech-Explanation Section @linnnsss 
      - [x] ckb-address
      - [x] Others..
         
-  [x] Better UX/UI
   - [x] Tooltips for terms on each page @yfeng2824 
   - [x] [Readability improvement](https://github.com/nervosnetwork/docs.nervos.org/issues/375) on UI @yfeng2824 
   - [x] Enhance Visual Accessibility
     - [x] For concepts in Tech explanation with flowchart https://github.com/nervosnetwork/docs.nervos.org/issues/240
         - [x] Cell
         - [x] Script
         - [x] WitnessArgs
     - [x] For `How CKB works` with intuitive concept images (Cell box vs UTXO box)
   - [x] Redirect dead links(maybe use https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-client-redirects) @RetricSu 
      - [x]  /docs/basics/guides/mining-ckb （直接跳转）
      - [x]  /docs/reference/halving (从旧文档整理出新的）
      - [x]  /docs/basics/guides/crypto wallets/neuron （integrate wallet)
    
  
  - [ ] Research(optional) @RetricSu 
     - [ ] Chat with LLM based on RFC docs

**Comment by RetricSu** [comment]
### V3 Menu Structure

- Getting Started
  - How CKB Works
  - Quick Start(5m)
  - Blockchain Networks
  - CKB vs BTC difference
  - UTXO vs Account Model
 
- SDK & Devtools
  - Rust
  - Go
  - Java
  - Javascript/Typescript (Lumos)
  - CommandLine Tool (CLI)
  - Other Dev Tools

- DApp Tutorials
  - Transfer CKB
  - Store Data on Cell
  - Create a Fungible Token
  - Create a DOB
  - Build a Simple Lock
  - Build a Simple Dex

- Scripts(Smart Contract)
  - How Script Works
  - Language Choices for Script
  - CKB syscall for Script
  - Script Cycles Limits
  - Tutorial: Minimal Script
  - Tutorial: User Defined Token(UDT) 
  - Tutorial: Type ID (Upgradable Script)
  - Tutorial: Run Javascript on CKB
  - Tutorial: Script With Dynamic Linking
  - Tutorial: Call Script from Script with Spawn
  - Compile Script
  - Debug and Testing Script
  - Deploy Script to CKB
  - Script Error Code
  
- Integrate Wallets
  - Select Your Wallets
     - Unisat/Metamask
  - All-In-One With `CCC` connector
  - ...
    
- Common Scripts 
  - Omnilock
      - How this Script works
      - Script deployed info
      - Offchain SDK examples
      - On-chain integration examples
  - Xudt
  - DAO  

- Network & Nodes
  - RPCs
  - Run a Mainnet Node
  - Run a Testnet Node
  - Run a Devnet Node
  - Run a Public RPC Node

- Molecule(Serialization)
  - Why do I need this? 
  - How Molecule Works
  - Example
  - ...
	
- How-to Guide
	- How to Sign transactions
	- How to use Since
	- How to calculate transaction hash
	- ...

- Tech Explanation
  - ....
  - Hash Type
  - Dep Group
  - Dep Cell
  - Cell Dep
  - Type Script
  - Lock Script
  - Script Args
  - ...
  - Concept Glossary
	
- Mining CKB

- Ecosystem
	- Projects
	- Code Repos
	- Communities(TBD) 

- Contribution
	- How to Contribute
	- List of Tasks
	- Code of Conduct

- Resources
  - Position Paper
  - CKB RFCs
  - CKB Dev Log
  - Other Docs
  - Other Resources
  - ...
